dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/managedobject/CDIManagedObjectFactoryImpl.java

```diff
@@ -116,4 +123,10 @@ public ManagedObjectContext createContext(boolean nonContextual) throws ManagedO
     public String toString() {
         return "CDI Managed Object Factory for class: " + getManagedObjectClass().getName();
     }
+
+    private void isWeldProxy() {
+        if (getManagedObjectClass().getSimpleName().contains("Proxy$_$$_WeldSubclass")) {
```

Comment:
does Weld only use `"Proxy$_$$_WeldSubclass"` as the extension?  In at least one test case that we've run across, we saw:  `com.ibm.ws.jaxrs20.cdi12.fat.loadonstartup.StartupFilter1$Proxy$_$$_WeldClientProxy`.

Maybe this should be:
```suggestion
        if (getManagedObjectClass().getSimpleName().contains("Proxy$_$$_Weld")) {
```

Reply:
Good catch.